apps/wallet-mobile/src/features/Swap/common/useSwapTx.ts

```diff
@@ -9,11 +8,19 @@ import {YoroiEntry, YoroiUnsignedTx} from '../../../yoroi-wallets/types'
 
 export const useSwapTx = (options?: UseMutationOptions<YoroiUnsignedTx, Error, {entry: YoroiEntry; datum: Datum}>) => {
   const {createOrder} = useSwap()
-  const orderMetadata = omit(createOrder, ['datumHash', 'datum'])
+
   const metadata = [
     {
       label: '674',
-      data: {msg: [`${createOrder.selectedPool?.provider}: Swap B for A Order Request`], order: {...orderMetadata}},
+      data: {
+        provider: [createOrder.selectedPool.provider],
+        sellTokenId: splitStringInto64CharArray(createOrder.amounts.sell.tokenId),
+        sellQuantity: [createOrder.amounts.sell.quantity],
+        buyTokenId: splitStringInto64CharArray(createOrder.amounts.buy.tokenId),
+        buyQuantity: [createOrder.amounts.buy.quantity],
+        depositFee: [createOrder.selectedPool.deposit.quantity],
+        poolId: [splitStringInto64CharArray(createOrder.selectedPool.poolId)],
+      },
```

Comment:
I think it needs a minor update according to https://developers.cardano.org/docs/governance/cardano-improvement-proposals/cip-0020/
```suggestion
      data: {
        msg: splitStringInto64CharArray(JSON.stringify({provider: ..,sellTokenId: ...})),
      },
```

Reply:
the shape should have the `msg` key and it should be string, or an array of 64bytes of strings.